src/app/test/service/testRunner.ts

```diff
@@ -43,18 +44,19 @@ export async function runTests(testSession: TestSession, filter?: string[]) {
   return finalizeTestSession(testSession)
 }
 
-export async function runTest(testSession: TestSession, test: TestCase): Promise<DetailedTestResult> {
+export async function runTest(testSession: TestSession, testCase: TestCase): Promise<DetailedTestResult> {
+  httpClient.defaults.lookup = testCase.dnsLookup
```

Comment:
`httpClient.defaults.lookup = undefined` and `httpClient.defaults.lookup = testCase.dnsLookup` should be at the same place, otherwise it will be hard keep the code organized. This PR has places with `httpClient.defaults.lookup = undefined` and it is not clear to the reader why. 

Edit: On a second thought, I'd much prefer to see a `before` and `after` functions for each test case rather than `dnsLookup` function. It serves more general purpose, the `runTest` function should not care about `dnsLookup` of `httpClient` but it can run `before` and `after` of each test case. When we add IPv6 tests, instead of overriding `dnsLookup` function, it would be more readable if we directly wrote `httpClient.defaults.lookup = () => { ...}` in `before` field and `httpClient.defaults.lookup = undefined` in `after` field, putting these two related code in the same context.